runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/SchemaTranslation.java

```diff
@@ -139,32 +150,43 @@ private static Field fieldFromProto(RunnerApi.Schema.Field protoField) {
   }
 
   private static FieldType fieldTypeFromProto(RunnerApi.Schema.FieldType protoFieldType) {
-    TypeName typeName = TYPE_NAME_MAPPING.inverse().get(protoFieldType.getTypeName());
     FieldType fieldType;
-    switch (typeName) {
-      case ROW:
-        fieldType = FieldType.row(fromProto(protoFieldType.getRowSchema()));
+    switch (protoFieldType.getTypeInfoCase()) {
```

Comment:
Another not-for-this PR comment that this would be cleaner with the `switch` in a function so the branches could all `return`.